src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -127,13 +130,18 @@ public synchronized void launch(SlaveComputer computer, TaskListener listener) {
 
             TaskListener runListener = template.getListener();
 
-            LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podName });
+            LOGGER.log(FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podName });
             try {
                 pod = client.pods().inNamespace(namespace).create(pod);
             } catch (KubernetesClientException e) {
-                if (e.getMessage() != null && e.getMessage().contains("pod rejected due to") && e.getMessage().contains("openpolicyagent")) {
+                String k8sCode = Integer.toString(e.getCode());
```

Comment:
String conversion needed? comparing just against the integer like the following snippet does not work?

```java
int code = e.getCode();
if (code >= 400 && code < 500) {
   // cancel the job
else if (code >= 500 && code < 600) {
   // retry
} else {
   // retry
}
```

Reply:
Updated.